git/git_filter_protocol.go

```diff
@@ -36,103 +38,75 @@ func NewObjectScanner(r io.Reader, w io.Writer) *ObjectScanner {
 	}
 }
 
-func (o *ObjectScanner) Init() bool {
+func (o *ObjectScanner) Init() error {
 	tracerx.Printf("Initialize filter")
 	reqVer := "version=2"
 
 	initMsg, err := o.p.readPacketText()
 	if err != nil {
-		fmt.Fprintf(os.Stderr,
-			"Error: reading filter initialization failed with %s\n", err)
-		return false
+		return fmt.Errorf("reading filter initialization failed with %s", err)
```

Comment:
Big :+1: for using errors and not a return :ghost: bool here. Any thoughts on using error wrapping instead? It doesn't seem like this would be clobbering any interesting error info though...